test/unit/pool/basic-tests.js

```diff
@@ -210,6 +211,66 @@ test('pool#connect client.poolCount', function() {
         assert.equal(client.poolCount, undefined,
           'after pool is destroyed, count should be undefined');
       });
-    })
+    });
+  });
+});
+
+pg.defaults.block = false;
+pg.defaults.poolSize = 1;
+
+test('pool#connect acquire errors if the pool is full', function() {
```

Comment:
Also - there are more tests around the `generic-pool` implementation then there are here, this just covers the common cases.